### ISSUE TITLE
Small issue with trailing modifier

### ISSUE THREAD

**Issue by garronej** [typehero/type-route#89]
Hi @bradenhs,  

We have a small issue with trailing modifier, nothing too serious but I think it is worth fixing.  

Let's say we have this route
```typescript
defineRoute(
  {
    slug: param.path.trailing.optional.string
  },
  p => `/foo/${p.slug}`
)
```

Now if the url of my page is `example.com/foo/xxx`

`page.params.slug` will be `/xxx`  I would have expected `xxx`.  

Now if I do `pages.myPage({ "slug": "xxx" }).replace()`, the url in the browser will still be
`example.com/foo/xxx` but `page.params.slug` will give me `xxx`.  

I think it's a problem that we can't strictly predict what the `route` object will be by looking at the URL.  

Best regards

**Comment by bradenhs** [comment]
Ah yes. This does seem like an issue. I'll take a closer look. Thank you for reporting!

**Comment by bradenhs** [comment]
Should be fixed now in 0.7.0 https://github.com/typehero/type-route/releases/tag/0.7.0

**Comment by garronej** [comment]
Great! An update!
